workers.go

```diff
@@ -186,19 +227,31 @@ func (api *API) UploadWorker(requestParams *WorkerRequestParams, data string) (W
 	return r, nil
 }
 
-// UploadWorkerWithName push raw script content for your worker
-// This is an enterprise only feature https://developers.cloudflare.com/workers/api/config-api-for-enterprise/
-//
-// API reference: https://api.cloudflare.com/#worker-script-upload-worker
-func (api *API) uploadWorkerWithName(scriptName string, data string) (WorkerScriptResponse, error) {
-	if api.OrganizationID == "" {
-		return WorkerScriptResponse{}, errors.New("organization ID required for enterprise only request")
+func (api *API) doMultipartUploadWorker(uri, data string, metadata WorkerMetadataParams) (WorkerScriptResponse, error) {
+	var r WorkerScriptResponse
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	metadataPart, err := writer.CreateFormFile("metadata", "metadata.json")
```

Comment:
we should probably not hardcode these as  `metadata.json` and `script.js` are not the only available options 
https://developers.cloudflare.com/workers/api/resource-bindings/webassembly-modules/